### PR TITLE
Added optional serde support

### PR DESCRIPTION
**Pull request by nodeSpace**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -313,9 +327,11 @@ where
 /// let mut engine: SimSearch<usize> = SimSearch::new_with(
 ///     SearchOptions::new().case_sensitive(true));
 /// ```
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct SearchOptions {
     case_sensitive: bool,
     stop_whitespace: bool,
+    #[cfg_attr(feature = "serde", serde(skip))]
     stop_words: &'static [&'static str],
```
Comment (andylokandy) [inline src/lib.rs:335]: `stop_words` can be rewritten in `Vec<String>` so as to support serialization.

Comment (nodeSpace) [reply src/lib.rs]: Ah, that's a much better way of handling it; committed changes

File: Cargo.toml [hunk 2]
```diff
@@ -27,6 +33,9 @@ quickcheck_macros = "0.8"
 itertools = "0.8"
 criterion = "0.2.11"
 
+[dev-dependencies.serde]
+version = "1.0"
```
Comment (andylokandy) [inline Cargo.toml:37]: What is this for?

Comment (nodeSpace) [reply Cargo.toml:37]: seems its not needed, its leftover from when I was figuring something out.

**Comment by andylokandy** [comment]
@nodeSpace Thank you!

**Pull request merged**